inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/HexFunction.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sdk.transform.process.function;
+
+import org.apache.inlong.sdk.transform.decode.SourceData;
+import org.apache.inlong.sdk.transform.process.Context;
+import org.apache.inlong.sdk.transform.process.operator.OperatorTools;
+import org.apache.inlong.sdk.transform.process.parser.ValueParser;
+
+import net.sf.jsqlparser.expression.Function;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Optional;
+
+/**
+ * HexFunction
+ * description: If the input argument is a numeric value (such as an integer), the HEX function converts the value to the corresponding hexadecimal string.
+ *              If the input argument is a string, the HEX function converts each character in the string to its corresponding hexadecimal ASCII encoding and returns the hexadecimal representation of the entire string.
+ */
+public class HexFunction implements ValueParser {
+
+    private ValueParser valueParser;
+
+    public HexFunction(Function expr) {
+        valueParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
+    }
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object valueObj = valueParser.parse(sourceData, rowIndex, context);
+        try {
+            return hex(OperatorTools.parseBigDecimal(valueObj)).toUpperCase();
+        } catch (NumberFormatException e) {
+            return hex(OperatorTools.parseString(valueObj));
```

Comment:
use `instanceof` to derive the specified method call, instead of the exception

Reply:
> use `instanceof` to derive the specified method call, instead of the exception
I have tried that before, but if we use `instanceof` to derive the specified method call like this
```
if (valueObj instanceof BigDecimal) {
            return hex(OperatorTools.parseBigDecimal(valueObj)).toUpperCase();
        } else if(valueObj instanceof String) {
            return hex(OperatorTools.parseString(valueObj)).toUpperCase();
        }
        return null;
```
because `valueParser.parse(sourceData, rowIndex, context);` always return a String so `if (valueObj instanceof BigDecimal)` will never reached, and i checked the `parseBigDecimal` method, it seens that it also doesn't handle the NumberFormatException, i think i should handle it here or the `parseBigDecimal` method

Reply:
Maybe you can make a new method called `isBigDecimal` to check the type of input valueObj, if true, parse it in BigDecimal way; otherwise, parse it as String.

```
if (isBigDecimal(valueObj)) {
     return hex(OperatorTools.parseBigDecimal(valueObj)).toUpperCase();
} 
return hex(OperatorTools.parseString(valueObj)).toUpperCase();
```

Reply:
> Maybe you can make a new method called `isBigDecimal` to check the type of input valueObj, if true, parse it in BigDecimal way; otherwise, parse it as String.
> 
> ```
> if (isBigDecimal(valueObj)) {
>      return hex(OperatorTools.parseBigDecimal(valueObj)).toUpperCase();
> } 
> return hex(OperatorTools.parseString(valueObj)).toUpperCase();
> ```

i fix it as using regex to check the data

Reply:
done